### DISCUSSION TITLE
Check if put_object was successful

### DISCUSSION THREAD

**Discussion by pamaku** [awslabs/aws-sdk-rust#214] [Q&A] [answered]
Rust newbiew and need help on how to check if PutObject was successful. 

async fn s3_post(pdf: Vec<u8>, key: &String) -> Result<(), Box<dyn Error>> {
    let client = s3::Client::from_env();
    let body = ByteStream::from(pdf);
    let resp =  client
        .put_object()
        .bucket("mybucket")
        .key(key)
        .body(body)
        .send()
        .await?;

        match resp{
            Ok(success) => success,
            Error(why) => println!("No cigar")
        }

    Ok(())
}

Ok and Error arms of match throwing errors

Ok - expected struct `aws_sdk_s3::output::PutObjectOutput`, found enum `std::result::Result`
Error - expected tuple struct or tuple variant, found trait `Error`

Any pointers appreciated.

**Comment by rcoh** [comment, marked as answer]
The `?` is equivalent to "return if error". If you remove it your code should compile. 

https://doc.rust-lang.org/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator

**Reply by pamaku** [reply]
Error arm still  has issue

expected tuple struct or tuple variant, found trait `Error`

not a tuple struct or tuple variantrustc(E0532)
